superduperdb/vector_search/atlas.py

```diff
@@ -0,0 +1,141 @@
+import json
+import re
+import typing as t
+from functools import cached_property
+
+import pymongo
+
+from superduperdb import CFG
+from superduperdb.backends.mongodb.query import Aggregate
+from superduperdb.vector_search.base import BaseVectorSearcher
+
+if t.TYPE_CHECKING:
+    from superduperdb import VectorIndex
+    from superduperdb.base.datalayer import Datalayer
+
+
+class MongoAtlasVectorSearcher(BaseVectorSearcher):
+    """
+    Implementation of atlas vector search
+
+    :param identifier: Unique string identifier of index
+    """
+
+    def __init__(self, identifier: str, **kwargs):
+        self.identifier = identifier
+        self.conn = pymongo.MongoClient(CFG.cluster.vector_search)
+
+    def __len__(self):
+        pass
+
+    @cached_property
+    def index(self):
```

Comment:
We should implement this class so that there are no class variables `db`, `CFG`, `vector_index` etc.